internal/pkg/install/prepare.go

```diff
@@ -33,281 +62,260 @@ func Prepare(data *userdata.UserData) (err error) {
 		return nil
 	}
 
-	// Root Device Init
-	if data.Install.Root.Device == "" {
-		return errors.Errorf("%s", "install.rootdevice is required")
-	}
+	// Verify that the target device(s) can satisify the requested options.
 
-	if data.Install.Root.Size == 0 {
-		// Set to 1G default for funzies
-		data.Install.Root.Size = 2048 * 1000 * 1000
+	if err = VerifyRootDevice(data); err != nil {
+		return errors.Wrap(err, "failed to prepare root device")
 	}
-
-	if len(data.Install.Root.Data) == 0 {
-		// Should probably have a canonical location to fetch rootfs - github?/s3?
-		// need to figure out how to download latest instead of hardcoding
-		data.Install.Root.Data = append(data.Install.Root.Data, "https://github.com/autonomy/talos/releases/download/"+version.Tag+"/rootfs.tar.gz")
+	if err = VerifyDataDevice(data); err != nil {
+		return errors.Wrap(err, "failed to prepare data device")
 	}
-
-	// Data Device Init
-	if data.Install.Data.Device == "" {
-		data.Install.Data.Device = data.Install.Root.Device
+	if err = VerifyBootDevice(data); err != nil {
+		return errors.Wrap(err, "failed to prepare boot device")
 	}
 
-	if data.Install.Data.Size == 0 {
-		// Set to 1G default for funzies
-		data.Install.Data.Size = 1024 * 1000 * 1000
+	manifest := NewManifest(data)
+
+	if data.Install.Wipe {
+		if err = WipeDevices(manifest); err != nil {
+			return errors.Wrap(err, "failed to wipe install device(s)")
+		}
 	}
 
-	// Boot Device Init
-	if data.Install.Boot != nil {
-		if data.Install.Boot.Device == "" {
-			data.Install.Boot.Device = data.Install.Root.Device
+	// Create and format all partitions.
+
+	for dev, targets := range manifest.Targets {
```

Comment:
Is it worth migrating this to a function as well -- `func PartitionDevice(manifest.Targets)`

Reply:
I think so. Updated to be in a dedicated `func`.